src/cargo/util/config.rs

```diff
@@ -32,22 +33,119 @@ use util::Filesystem;
 
 use self::ConfigValue as CV;
 
+#[derive(Clone, Debug)]
+pub struct CargoDirs {
+    pub home_dir: Filesystem,
+    pub current_dir: PathBuf,
+    pub cache_dir: Filesystem,
+    pub config_dir: Filesystem,
+    pub data_dir: PathBuf,
+    pub bin_dir: PathBuf,
```

Comment:
It would be great to add short docstrings here, to explain which directory stores which data.